scipy/optimize/_spectral.py

```diff
@@ -122,12 +131,21 @@ def fmerit(F):
         if callback is not None:
             callback(x_k, F_k)
 
+        # Check convergence
         if F_k_norm < ftol * F_0_norm + fatol:
             # Converged!
             message = "successful convergence"
             converged = True
             break
 
+        if f_k < last_improved_f:
```

Comment:
This probably should be something like `f_k < 0.99*last_improved_f` to enforce at least linear convergence...